### PR TITLE
update fhir net api version to 2.0.3

### PR DESCRIPTION
**Pull request by hongyh13**
update the fhir net sdk to 2.0.3, the breaking changes related to this project are (https://github.com/FirelyTeam/firely-net-sdk/wiki/Breaking-changes-in-2.0):

1. SimpleQuantity and MoneyQuantity (R4+) have been removed, and Quantity is used.
2. Type Primitive has been renamed to PrimitiveType.
3. The FhirPath engine is upgraded to support FhirPath normative. Equality, equivalence and ordering for (partial) datetimes has been tweaked to better deal with difference in precision. 
4. The "Primitives.GetNativeRepresentation" API has been removed.

### CODE REVIEW

**Review by moria97 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Processors/GeneralizeProcessorTests.cs [hunk 1]
```diff
@@ -67,11 +67,14 @@ public static IEnumerable<object[]> GetDateNodesToGeneralizeWithRangeMapping()
             yield return new object[] { new Date("1990-01-01"), "1990" };
             yield return new object[] { new Date("2000-01-01"), null, "Redact" };
             yield return new object[] { new Date("1990"), "1990", "Redact" };
-            yield return new object[] { new Date("2000"), null, "Redact" };
-            yield return new object[] { new Date("2010"), "2010-01-01", "Redact" };
-            yield return new object[] { new Date("2010-01-01"), null, "Redact" };
+            yield return new object[] { new Date("2000"), "1990", "Redact" };
+            yield return new object[] { new Date("2000-01-01"), null, "Redact" };
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Anonymizer.Shared.Core.UnitTests/Processors/GeneralizeProcessorTests.cs:71]: Seems duplicate with Line 69?

File: src/Microsoft.Health.Fhir.Anonymizer.R4.FunctionalTests/VersionSpecificTests.cs [hunk 2]
```diff
@@ -29,9 +30,14 @@ public static IEnumerable<object[]> GetR4OnlyResources()
...
         }
 
-        public static IEnumerable<object[]> GetCommonResourcesWithStu3OnlyField()
+        public static IEnumerable<object[]> GetCommonResourcesWithStu3OnlyValue()
         {
             yield return new object[] { "Stu3OnlyResource/Claim-Stu3.json" };
+
+        }
+
+        public static IEnumerable<object[]> GetCommonResourcesWithStu3OnlyElement()
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Anonymizer.R4.FunctionalTests/VersionSpecificTests.cs:39]: Just curious what's the difference between `value` and `element`?

**Review by BoyaWu10 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Anonymizer.R4.FunctionalTests/VersionSpecificTests.cs (hunk 2 above)
Comment (hongyh13) [reply src/Microsoft.Health.Fhir.Anonymizer.R4.FunctionalTests/VersionSpecificTests.cs:39]: > Just curious what's the difference between `value` and `element`?

'element' is the type e.g. 'Name', 'value' is the detail value e.g. 'Jason', the updated version returns different exception types for unsupported 'element' and 'value' format while the old version just returns the same exception, so we need to split the unit test now.

Comment (BoyaWu10) [reply src/Microsoft.Health.Fhir.Anonymizer.R4.FunctionalTests/VersionSpecificTests.cs:39]: Got it. Thanks!

**Review by tongwu-msft — APPROVED** [review]

**Review by sowu880 — APPROVED** [review]

**Review by brendankowitz — APPROVED** [review]
:shipit:

**Pull request merged**